src/StringContainsIgnoringWhiteSpace.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace WMDE\HamcrestHtml;
+use Hamcrest\Text\SubstringMatcher;
+
+class StringContainsIgnoringWhiteSpace extends SubstringMatcher
+{
+	public function __construct($substring)
+	{
+		parent::__construct($this->_stripSpace($substring));
+	}
+
+	/**
+	 * Matches if value is a string that contains $substring consider all whitespace as single space
+	 */
+	public static function containsStringIgnoringWhiteSpace($substring)
+	{
+		return new self($substring);
+	}
+
+	/**
+	 * @param \DOMElement $item
+	 *
+	 * @return bool
+	 */
+	protected function evalSubstringOf($item)
+	{
+		return (false !== strpos($this->_stripSpace((string) $item), $this->_substring));
+	}
+
+	protected function relationship()
+	{
+		return 'containing';
+	}
+
+	/**
+	 * Copied from IsEqualIgnoringWhiteSpace
+	 *
+	 * @param $string
+	 *
+	 * @return string
+	 */
+	private function _stripSpace($string)
+	{
+		$parts = preg_split("/[\r\n\t ]+/", $string);
```

Comment:
Is there a specific reason for this exact set of whitespace characters? I would prefer this to use `\s` (either with or without the `/u` Unicode flag – both might make sense, depending on what you intended to mean with "whitespace" in this classes description).

On the other hand, if this set is from, lets say, the HTML specification, please add a comment explaining this.